frontend/src/components/AddRepository.tsx

```diff
@@ -1,26 +1,58 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import "assets/styles/addRepository.scss";
-import { Input } from "antd";
+import { Input, Spin } from "antd";
 import { useNavigate } from "react-router-dom";
 
-import { searchRepositoryByUrl } from "api";
+import { createRepositoryByUrl, getFileTree, searchRepositoryByUrl } from "api";
+import { useDataSettingContext } from "context/DataSettingContext";
 
 const AddRepository: React.FC = () => {
   const [url, setUrl] = useState<string>("");
+  const [loadRepository, setLoadRepository] = useState<boolean>(false);
+  const { setRepository } = useDataSettingContext();
   const navigate = useNavigate();
 
-  const handleEnterPress = async () => {
-    const result = await searchRepositoryByUrl(url);
+  useEffect(() => {
+    //Implementing the setInterval method
+    if (loadRepository) {
+      let interval: any;
+
+      const fetchData = async () => {
+        const data = await createRepositoryByUrl(url);
+
+        let checkRepository: any;
+        interval = setInterval(async () => {
+          try {
+            checkRepository = await getFileTree(data.id); // FAKE REQUEST JUST TO TEST
```

Comment:
I suggest you add a function in the `api/repositories.ts` that queries the repository, instead of the files. I shipped https://github.com/visevol/Cscope/pull/114, so you can use the `last_synced_at` field on the JSON of the repository to know determine when the repository is ready. 

Something like: 
```typescript
export async function getRepository(id: number): Promise<Repository | null> {
  const response = await axios.get(`/repositories/${id}`);

  return response.status == 200 ? response.data : null;
}
``` 

It returns a payload like this:
```javascript
{
  id: 1,
  name: "rails",
  domain: "github.com",
  path: "/rails/rails",
  url: "https://github.com/rails/rails",
  created_at: "2024-12-02T03:25:31.654Z",
  updated_at: "2024-12-02T03:25:31.654Z",
  last_synced_at: null
}
```

^^^ Note that the `last_synced_at` is `null`.

Once the repository is ready, it will contain a date:

```javascript
{
  id: 1,
  name: "rails",
  domain: "github.com",
  path: "/rails/rails",
  url: "https://github.com/rails/rails",
  created_at: "2024-12-02T03:25:31.654Z",
  updated_at: "2024-12-02T03:27:02.606Z",
  last_synced_at: "2024-12-02T03:27:02.605Z"
}
```

^^ This is way faster, and you're guaranteed that the repository is READY to be consumed. 

The `getFileTree` function might return data, even if the repository is not ready to be consumed yet.

Reply:
It's done